File: AalsiTrader/Models/SigmaApproval.swift
import Foundation

struct SigmaApproval: Codable, Identifiable {
    var id: String { tradeId }
    let tradeId: String
    let symbol: String
    let signal: TradeSignal
    let entryPrice: Double
    let timestamp: ActivityTimestamp
    var status: ApprovalStatus
    var sigmaApprovedBy: String?
    var sigmaApprovedAt: ActivityTimestamp?
    var indicators: PaperTradeIndicators?
    var requiresApproval: Bool?

    var date: Date {
        timestamp.asDate
    }
}

enum ApprovalStatus: String, Codable {
    case pending
    case approved
    case rejected
}

File: AalsiTraderAndroid/app/src/main/java/com/aalsitrader/android/model/SigmaApproval.kt
package com.aalsitrader.android.model

import kotlinx.serialization.Serializable

@Serializable
data class SigmaApproval(
    val tradeId: String,
    val symbol: String = "",
    val signal: TradeSignal = TradeSignal.BUY,
    val entryPrice: Double = 0.0,
    val timestamp: ActivityTimestamp = ActivityTimestamp.Epoch(0.0),
    val status: ApprovalStatus = ApprovalStatus.pending,
    val sigmaApprovedBy: String? = null,
    val sigmaApprovedAt: ActivityTimestamp? = null,
    val indicators: PaperTradeIndicators? = null,
    val requiresApproval: Boolean? = null,
)

@Serializable
enum class ApprovalStatus { pending, approved, rejected }
